### PR TITLE
fix: Prevent saving with empty project name

### PR DESCRIPTION
**Pull request by Ash-exp**
# Description

Global config > Project > Prevent saving with empty project name

image

Fixes [AB 2060](https://dev.azure.com/DevtronLabs/Devtron/_boards/board/t/OSS%20adoption/Stories/?workitem=2060)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

1. Go to Global Config > Projects
2. Click on add project
3. Click on Save without providing a project name
4. Clicking on save without providing a Project name should show a field validation error with message "This is a required field"

- [x] Test Manually
- [ ] Test B

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/project/Project.tsx [hunk 1]
```diff
@@ -57,6 +62,33 @@ export class Project extends Component<ProjectProps, ProjectState>  {
...
+        else {
+            this.setState({
+                validationError: ""
+            })
+        }
+        event.preventDefault()
+        this.props.saveProject(this.props.index)
+    }
+
+    isProjectNameExists(): boolean {
```
Comment (shivani170) [inline src/components/project/Project.tsx:88]: Please move it to parent component

File: src/components/project/Project.tsx [hunk 2]
```diff
@@ -57,6 +62,33 @@ export class Project extends Component<ProjectProps, ProjectState>  {
         }
     }
 
+    saveProjectData = (event) => {
+        const _projectExists = this.isProjectNameExists()
+        if (!this.props.name) {
+            this.setState({
+                validationError: "This is a required field"
```
Comment (shivani170) [inline src/components/project/Project.tsx:69]: Please extract these messages from constants

File: src/components/project/Project.tsx [hunk 3]
```diff
@@ -87,49 +119,60 @@ export class Project extends Component<ProjectProps, ProjectState>  {
...
-                        Save
-                </ButtonWithLoader>
-                </div>
-                </div>
-            </form >
-        </div>
+        return (
+            <div>
+                <form
+                    onSubmit={(e) => {
```
Comment (shivani170) [inline src/components/project/Project.tsx:128]: 1. Avoid inline function
2. Can try to remove prop drilling instead can use this.props.index directly

File: src/components/project/Project.tsx [hunk 4]
```diff
@@ -87,49 +119,60 @@ export class Project extends Component<ProjectProps, ProjectState>  {
...
+                            name="name"
+                            value={this.props.name}
+                            placeholder="e.g. My Project"
+                            className="form__input"
+                            autoFocus
+                            onChange={(event) => {
+                                this.props.handleChange(event, this.props.index, 'name')
+                            }}
+                        />
+                        {isValid.name ? null : <span className="form__error">{errorMessage.name}</span>}
```
Comment (shivani170) [inline src/components/project/Project.tsx:148]: Flip the condition

**Review by shivani170 — APPROVED** [review]

File: src/components/project/Project.tsx (hunk 1 above)
Comment (Ash-exp) [reply src/components/project/Project.tsx]: implemented 👍

**Review by shivani170 — APPROVED** [review]

**Review by vivek-devtron — APPROVED** [review]

**Pull request merged**